### ISSUE TITLE
Allow dynamic configuration

### ISSUE THREAD

**Issue by nicolasblanco** [sorentwo/braintree-elixir#37]
First of all, thank you for your library, it's working very well so far.

I'm working on an application of a white-brand marketplace and vendors can setup their own Braintree API keys in some back-office. For this application, I would like to be able to read the API keys from the database and pass them as parameters on every Braintree call instead of application-wide config.

Another developer has begun work on that here:
https://github.com/steady-media/braintree-elixir/commit/f5bf7367ad9e38acc63fe9b479cc5aab9c626811

Basically an optional `opts` list could be passed on every call and if it's there, it would then use the passed API keys and environment, otherwise it would then use the current behaviour of reading the application config.

What do you think of such approach? It should not break the current API...
I would gladly continue the work of @manukall and submit a PR.

Thanks.

**Comment by sorentwo** [comment]
@nicolasblanco I'm open to an approach like that, it makes sense to provide finer grained options for authentication and communications.

If you've started work on this please open a PR, otherwise I'll gladly tackle it myself.

**Comment by nicolasblanco** [comment]
@sorentwo : Thanks, I've made a PR here: https://github.com/sorentwo/braintree-elixir/pull/38

**Comment by sorentwo** [comment]
Closed with #746f236
